include/shark/Data/SparseData.h

```diff
@@ -116,36 +155,37 @@ SHARK_EXPORT_SYMBOL void importSparseData(
 	std::size_t batchSize = LabeledData<RealVector, unsigned int>::DefaultBatchSize
 );
 
+/// \brief Import regression data from a sparse data (libSVM) file.
+///
+/// \param  dataset       container storing the loaded data
+/// \param  fn            the file to be read from
+/// \param  highestIndex  highest feature index, or 0 for auto-detection
+/// \param  batchSize     size of batch
+SHARK_EXPORT_SYMBOL void importSparseData(
+	LabeledData<CompressedRealVector, RealVector>& dataset,
+	std::string fn,
+	unsigned int highestIndex = 0,
+	std::size_t batchSize = LabeledData<RealVector, RealVector>::DefaultBatchSize
+);
+
 
-/// \brief Export data to sparse data (libSVM) format.
+/// \brief Export classification data to sparse data (libSVM) format.
 ///
 /// \param  dataset     Container storing the  data
-/// \param  fn          Output file
-/// \param  dense       Flag for using dense output format
+/// \param  stream      Output stream
+/// \param  dense       Flag for writing all components, including zeros
 /// \param  oneMinusOne Flag for applying the transformation y<-2y-1 to binary labels
 /// \param  sortLabels  Flag for sorting data points according to labels
-/// \param  append      Flag for appending to the output file instead of overwriting it
 template<typename InputType>
-void exportSparseData(LabeledData<InputType, unsigned int>& dataset, const std::string &fn, bool dense=false, bool oneMinusOne = true, bool sortLabels = false, bool append = false) {
+void exportSparseData(LabeledData<InputType, unsigned int> const& dataset, std::ostream& stream, bool dense=false, bool oneMinusOne = true, bool sortLabels = false)
+{
 	std::size_t elements = dataset.numberOfElements();
-    std::ofstream ofs;
-    
-    // shall we append only or overwrite?
-    if (append == true) {
-        ofs.open (fn.c_str(), std::fstream::out | std::fstream::app );
-    } else {
-        ofs.open (fn.c_str());
-    }
-    
-	if( !ofs ) {
-		throw( SHARKEXCEPTION( "[exportSparseData] file can not be opened for writing" ) );
-	}
 
 	size_t dim = inputDimension(dataset);
 	if(numberOfClasses(dataset)!=2) oneMinusOne = false;
 
 	std::vector<detail::LabelSortPair> L;
-	if(sortLabels) {
+	if (sortLabels) {
 		for(std::size_t i = 0; i < elements; i++) 
 			L.push_back(detail::LabelSortPair(dataset.element(i).label, i));
```

Comment:
>>
>>   	size_t dim = inputDimension(dataset);
>>   	if(numberOfClasses(dataset)!=2) oneMinusOne = false;
>>
>>   	std::vector<detail::LabelSortPair> L;
>> -	if(sortLabels) {
>> +	if (sortLabels) {
>>   		for(std::size_t i = 0; i < elements; i++)
>>   			L.push_back(detail::LabelSortPair(dataset.element(i).label, i));
>
> When we are at it, LabelSortPair can be replaced by a shark::KeyValuePair<unsigned int,double> found in Core/utility/KeyValuePair.h and instead of the constructor one can use makeKeyValuePair(dataset.element(i).label, i)

I will replace it with a proper C++11 solution that also
gets along without element access. I see, there is more to
fix than the missing regression support.